src/commands/heroku.rs

```diff
@@ -1,36 +1,131 @@
 
 use serenity::framework::standard::{macros::command, Args, CommandResult};
 use serenity::model::prelude::*;
 use serenity::prelude::*;
 
+use crate::authorizations::users::*;
 use crate::config::Config;
 
 #[derive(Debug, Deserialize)]
 struct HerokuApp {
     id: String,
     name: String,
+    released_at: String,
+    web_url: String,
+}
+
+// Get app by name or id
+#[command]
+pub fn get_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandResult {
+    let config = bot_config(ctx);
+
+    let app_name = args.single::<String>().unwrap();
+
+    let response = heroku_client(&config.heroku_api_key)
+        .get()
+        .apps()
+        .app_name(&app_name)
+        .execute::<HerokuApp>();
+
+    if is_authorized(&msg.author.id.to_string(), &*config) {
+        msg.reply(
+            ctx,
+            match response {
+                Ok((_, _, Some(app))) => app_response(app),
+                Ok((_, _, None)) => "There is no Heroku app by that name".into(),
+                Err(err) => {
+                    println!("Err {}", err);
+                    "An error occured while fetching your Heroku app".into()
+                }
+            },
+        )?;
+    } else {
+        msg.reply(
+            ctx,
+            format!(
+                "{}'s Discord ID is not authorized to run this command",
+                msg.author.name
+            ),
+        )?;
+    }
+
+    Ok(())
 }
 
 #[command]
 pub fn get_apps(ctx: &mut Context, msg: &Message, _args: Args) -> CommandResult {
-    let config = ctx.data.read().get::<Config>().expect("Expected config").clone();
+    let config = bot_config(ctx);
 
     let response = heroku_client(&config.heroku_api_key)
         .get()
         .apps()
         .execute::<Vec<HerokuApp>>();
 
-    msg.reply(ctx, match response {
-        Ok((_, _, Some(apps))) => app_response(apps),
-        Ok((_, _, None)) => "You have no Heroku apps".into(),
-        Err(err) => {
-            println!("Err {}", err);
-            "An error occured while fetching your Heroku apps".into()
-        }
-    })?;
+    if is_authorized(&msg.author.id.to_string(), &*config) {
+        msg.reply(
+            ctx,
+            match response {
+                Ok((_, _, Some(apps))) => apps_response(apps),
+                Ok((_, _, None)) => "You have no Heroku apps".into(),
+                Err(err) => {
+                    println!("Err {}", err);
+                    "An error occured while fetching your Heroku apps".into()
+                }
+            },
+        )?;
+    } else {
+        msg.reply(
+            ctx,
+            format!(
+                "{}'s Discord ID is not authorized to run this command",
+                msg.author.name
+            ),
+        )?;
+    }
+
+    Ok(())
+}
+
+#[command]
+pub fn restart_app(ctx: &mut Context, msg: &Message, mut args: Args) -> CommandResult {
+    let config = bot_config(ctx);
+
+    let app_name = args.single::<String>().unwrap();
```

Comment:
Same here – if the `unwrap()` can panic because of user errors, we should handle these errors.